VM/src/lbaselib.cpp

```diff
@@ -449,7 +449,8 @@ LUALIB_API int luaopen_base(lua_State* L)
 
     /* open lib into global table */
     luaL_register(L, "_G", base_funcs);
-    lua_pushliteral(L, "Luau");
+    lua_pushliteral(L, "Luau 0.503"); /* set _VERSION to display the version of Luau currently in use */
```

Comment:
Could you change this to:

```cpp
lua_pushliteral(L, LUAU_VERSION);
```

and add a string literal in `lua.h`?

e.g. 
```cpp
// classic style
#define LUAU_VERSION_MAJOR "0"
#define LUAU_VERSION_MINOR "503"

#define LUAU_VERSION "Luau " LUAU_VERSION_MAJOR "." 

// OR modern style
constexpr auto LUAU_VERSION_MAJOR = "0";
constexpr auto LUAU_VERSION_MINOR = "503";

constexpr LUAU_VERSION = "Luau " + LUAU_VERSION_MAJOR + "." + LUAU_VERSION_MINOR;
```

Reply:
yeah I suppose but I'd still prefer if a separate file (preferably .txt) could be used to track the version for ease of access when updating the version. To make it easier for new people to view the version without having to dig through the C++ code or start running luau. But that may be just another pull request later instead.

btw I tried the classic style as I have no idea how to get the modern style to work and this occured
<img width="261" alt="Screenshot 2021-11-08 at 13 31 17" src="https://user-images.githubusercontent.com/68239467/140750629-cafb5a69-88ef-4009-9ac6-c8210b33dcb3.png">
<img width="373" alt="Screenshot 2021-11-08 at 13 31 40" src="https://user-images.githubusercontent.com/68239467/140750674-f20e7077-594c-4356-a4ef-63a8bc7ba708.png">

Reply:
Actually I just realised that a version.txt file may not work properly with the binaries unless the value is saved during compiling